internal/plugin/metrics_plugin.go

```diff
@@ -186,12 +190,14 @@ func (m *Metrics) startExporters(ctx context.Context) {
 
 // Produces metrics from the MetricsProducer at the configured interval and pushes to the given channel.
 func (m *Metrics) runProducer(ctx context.Context, producer model.MetricsProducer) {
-	t := time.NewTicker(m.collectInterval)
+	m.metricsTimer = time.NewTicker(m.collectInterval)
+
+	defer m.metricsTimer.Stop()
```

Comment:
I'd like to better understand what the intention here was. So now instead of each producer having its own ticker, we are saving the ticker of the last started producer to `m.metricsTimer` and stopping our global ticker if **one** producer goes out of scope.

If the above behavior is what we want (i.e. all producers are tied to one another) then let's at least init the ticker one level higher, and potentially pass it as a parameter to `runProducer`, instead of in `runProducer`, which will be run multiple times.

I would think it would be better that producers are independent from one another but maybe there's something I haven't thought of.

Reply:
I didn't quite get that each producer had it's own timer here. If that's the case, I think the producer should own the timer? The metrics timer is specific to the metrics plugin in the current code, I was trying to tidy it up but I can revert for now until we decide on one way or another